ops/log.py

```diff
@@ -14,6 +14,7 @@
 
 """Interface to emit messages to the Juju logging system."""
 
+import contextvars
```

Comment:
Theoretically, (as I think you already pointed out somewhere), this could cause issues now, if someone put a logging call inside of the `juju_log` method, for example. I wonder if perhaps it could be good to strip out the changes in this module, which are (I think) very self-contained, and put them in a separate PR that could be merged quickly, to reduce the size of this one?